redash_toolbelt/examples/find-table-names.py

```diff
@@ -29,18 +22,31 @@ def find_table_names(url, key, data_source_id):
     ]
 
     tables_by_qry = {
-        query["id"]: [
-            match
-            for match in re.findall(PATTERN, query["query"])
-            if match in schema_tables or len(schema_tables) == 0
-        ]
+        query["id"]: extract_table_names(query["query"], schema_tables)
         for query in queries
-        if re.search(PATTERN, query["query"])
     }
 
     return tables_by_qry
 
 
+def extract_table_names(str_sql, schema_tables=[]):
```

Comment:
This doesn't just extract table names, it also filters them according to the schema tables.
This function could be simpler to understand if the filtering logic is left outside.

Reply:
I refactored for simplicity.